cmd/ctr/commands/client.go

```diff
@@ -50,17 +54,31 @@ func AppContext(context *cli.Context) (gocontext.Context, gocontext.CancelFunc)
 		log.L.Debugf("Using SOURCE_DATE_EPOCH: %v", tm)
 		ctx = epoch.WithSourceDateEpoch(ctx, tm)
 	}
+
 	return ctx, cancel
 }
 
 // NewClient returns a new containerd client
 func NewClient(context *cli.Context, opts ...containerd.ClientOpt) (*containerd.Client, gocontext.Context, gocontext.CancelFunc, error) {
 	timeoutOpt := containerd.WithTimeout(context.GlobalDuration("connect-timeout"))
-	opts = append(opts, timeoutOpt)
+	opts = append(opts,
+		timeoutOpt,
+		containerd.WithDialOpts([]grpc.DialOption{
+			// containerd's gRPC connections don't have credentials regardless of OpenTelemetry.
+			grpc.WithTransportCredentials(insecure.NewCredentials()),
+			grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor()),
```

Comment:
Should we check if `opentelemetry` or `tracing` flag is set before adding these options.

Reply:
Yeah. Will do.

Reply:
Done.

Reply:
Done.